src/lambda/eventConsumer/userEvents/userEventConsumer.ts

```diff
@@ -0,0 +1,72 @@
+import { UserEventHandler } from '../../snowplow/userEventHandler';
+import {
+  EventTypeString,
+  UserEventPayloadSnowplow,
+} from '../../snowplow/types';
+import { SQSRecord } from 'aws-lambda';
+
+export type UserEventPayload = {
+  userId: string;
+  email: string;
+  isPremium?: boolean;
+  apiId: string;
+  hashedId?: string;
+  guid?: number;
+  hashedGuid?: string;
+  name?: string;
+  isNative?: boolean;
+  isTrusted?: boolean;
+  clientVersion?: string;
+  language?: string;
+  snowplowDomainUserId?: string;
+  ipAddress?: string;
+  userAgent?: string;
+};
+
+//detail-type in event rule
+//defined here:
+// https://github.com/Pocket/pocket-event-bridge/blob/f462cbb1b166d937fcd62319f88c90efc7653ebc/.aws/src/event-rules/user-api-events/eventConfig.ts#L3
+export const DetailTypeToSnowplowMap: Record<string, EventTypeString> = {
+  'account-deletion': 'ACCOUNT_DELETE',
+  'account-email-updated': 'ACCOUNT_EMAIL_UPDATED',
+};
+
+export async function userEventConsumer(record: SQSRecord) {
+  await new UserEventHandler().process(getUserEventPayload(record.body));
+}
+
+export function getUserEventPayload(eventObj: any): UserEventPayloadSnowplow {
+  eventObj = JSON.parse(eventObj);
+  const message = JSON.parse(eventObj.Message);
+  const messageBody: UserEventPayload = message['detail'];
+  const detailType = message['detail-type'];
+
+  return {
+    user: {
+      id: messageBody.userId,
+      email: messageBody.email,
+      isPremium: messageBody.isPremium ? true : false, //set as 0 in payload
+      ...(messageBody.hashedId ? { hashedId: messageBody.hashedId } : {}),
+      ...(messageBody.guid ? { guid: messageBody.guid } : {}),
+      ...(messageBody.hashedGuid ? { hashedGuid: messageBody.hashedGuid } : {}),
+    },
+    apiUser: {
+      apiId: messageBody.apiId,
+      ...(messageBody.name ? { name: messageBody.name } : {}),
+      ...(messageBody.isNative ? { isNative: messageBody.isNative } : {}),
```

Comment:
this is a boolean value, right? If it's false, then it doesn't get included. I think the intention is to only include keys with a non-null value?

Reply:
yes, for booleans it doesn't include `false` and i was assuming default would be false. if its true, it gets passed along to snowplow

if i didn't do it this way, the test gets annoying to not include the fields even though the fields are marked optional